src/arduino/pressure_controller.cpp

```diff
@@ -0,0 +1,254 @@
+/*=============================================================================
+ * @file pressure_control.h
+ *
+ * COVID Respirator
+ *
+ * @section copyright Copyright
+ *
+ * Makers For Life
+ *
+ * @section descr File description
+ *
+ * This file implements the pressure control algorithm
+ */
+
+// INCLUDES ===================================================================
+
+// Associated header
+#include "pressure_controller.h"
+
+// Internal libraries
+#include "config.h"
+#include "debug.h"
+
+// INITIALISATION =============================================================
+
+PressureController pController(AirTransistor(), AirTransistor());
+
+// FUNCTION ===================================================================
+
+PressureController::PressureController(const AirTransistor &p_blower,
+                                       const AirTransistor &p_patient)
+    : m_cyclesPerMinuteCommand(20), m_minPeepCommand(5),
+      m_maxPlateauPressureCommand(30), m_apertureCommand(ANGLE_OUVERTURE_MAXI),
+      m_cyclesPerMinute(20), m_aperture(ANGLE_OUVERTURE_MAXI),
+      m_maxPeakPressure(60), m_maxPlateauPressure(30), m_minPeep(5),
+      m_currentPressure(-1), m_peakPressure(-1), m_plateauPressure(-1),
+      m_peep(-1), m_phase(CyclePhases::INHALATION), m_blower(p_blower),
+      m_patient(p_patient)
+
+{
+  computeCentiSecParameters();
+}
+
+void PressureController::setup()
+{
+  m_blower.actuator.attach(PIN_SERVO_BLOWER);
+  m_patient.actuator.attach(PIN_SERVO_PATIENT);
+
+  DBG_DO(Serial.print("mise en secu initiale");)
+
+  m_blower.actuator.write(m_blower.failsafe);
+  m_patient.actuator.write(m_patient.failsafe);
+}
+
+void PressureController::initLoop() {
+  m_peakPressure = -1;
+  m_plateauPressure = -1;
+  m_peep = -1;
+  m_phase = CyclePhases::INHALATION;
+  m_blower.reset();
```

Comment:
What is the conceptual difference between `m_blower.reset()` and `m_blower.actuator.write(m_blower.failsafe)`?

Reply:
In the original code, there were "secu" (=failsafe) command values so I kept them, while the default command may be different. Also, `m_blower.actuator.write(m_blower.failsafe)` actually sends the command to the actuator